pkg/frontend/querymiddleware/split_by_instant_interval.go

```diff
@@ -23,9 +23,11 @@ import (
 )
 
 const (
-	skippedReasonParsingFailed = "parsing-failed"
-	skippedReasonMappingFailed = "mapping-failed"
-	skippedReasonNoop          = "noop"
+	skippedReasonParsingFailed     = "parsing-failed"
+	skippedReasonMappingFailed     = "mapping-failed"
+	skippedReasonNoopSmallInterval = "noop-small-interval"
+	skippedReasonNoopSubquery      = "noop-subquery"
+	skippedReasonNoopNonSplittable = "noop-non-splittable"
```

Comment:
[nit] I would remove "noop" from the naming here (both the variable name and the value). These are the reasons why it was skipped, so I think "noop" prefix is not adding much value.

Reply:
Addressed in: https://github.com/grafana/mimir/pull/2642/commits/281a20ad817f41ecd5202797cd31cc296b9565da